src/query/util/execution/parallel_test.go

```diff
@@ -78,13 +79,34 @@ func TestOrderedParallel(t *testing.T) {
 }
 
 func TestSingleError(t *testing.T) {
+	defer leaktest.Check(t)()
 	requests := make([]Request, 3)
 	signalChan := make(chan bool)
-	requests[0] = &request{order: 0, wait: signalChan}
+
+	var cancelErr error
+	requests[0] = funcRequest(func(ctx context.Context) error {
+		// wait for the second goroutine to finish
+		<-signalChan
+
+		// wait for cancellation. This will hang if there's a bug here (i.e. the context doesn't get cancelled)
```

Comment:
Any way we can have it detect this case without waiting for the timeout to kill test execution?

Reply:
Yeah we can add an explicit timeout here, but I don't know that we need to. `leaktest` will fail the test if we do hang. I'll add a comment explaining as much.